drivers/serial/uart_stm32.c

```diff
@@ -93,20 +101,26 @@ static void uart_stm32_pm_policy_state_lock_put(const struct device *dev)
 }
 #endif /* CONFIG_PM */
 
-static inline void uart_stm32_set_baudrate(const struct device *dev,
-					   uint32_t baud_rate)
+static inline void uart_stm32_set_baudrate(const struct device *dev, uint32_t baud_rate)
 {
 	const struct uart_stm32_config *config = dev->config;
 	struct uart_stm32_data *data = dev->data;
 
 	uint32_t clock_rate;
 
 	/* Get clock rate */
-	if (clock_control_get_rate(data->clock,
-			       (clock_control_subsys_t *)&config->pclken,
-			       &clock_rate) < 0) {
-		LOG_ERR("Failed call clock_control_get_rate");
-		return;
+	if (IS_ENABLED(STM32_UART_OPT_CLOCK_SUPPORT) && (config->pclk_len > 1)) {
+		if (clock_control_get_rate(data->clock,
+				       (clock_control_subsys_t)&config->pclken[1],
+					   &clock_rate) < 0) {
+			LOG_ERR("Failed call clock_control_get_rate(pclken[1])");
+		}
+	} else {
+		if (clock_control_get_rate(data->clock,
+				       (clock_control_subsys_t)&config->pclken[0],
+					   &clock_rate) < 0) {
+			LOG_ERR("Failed call clock_control_get_rate(pclken[0])");
+		}
 	}
```

Comment:
Maybe `__ASSERT(...)` would be better instead of `LOG_ERR`?

But the formatting is off and missing `return`:
```suggestion
	if (IS_ENABLED(STM32_UART_OPT_CLOCK_SUPPORT) && (config->pclk_len > 1)) {
		if (clock_control_get_rate(data->clock,
					   (clock_control_subsys_t)&config->pclken[1],
					   &clock_rate) < 0) {
			LOG_ERR("Failed call clock_control_get_rate(pclken[1])");
			return;
		}
	} else {
		if (clock_control_get_rate(data->clock,
					   (clock_control_subsys_t)&config->pclken[0],
					   &clock_rate) < 0) {
			LOG_ERR("Failed call clock_control_get_rate(pclken[0])");
			return;
		}
	}
```

Reply:
Many thanks for prompt review.
I also prefer assertions but unfortunately many projects have them disabled and in such case there is no trace about problem occurence.
@erwango What do you prefer `LOG_ERR` or `__ASSERT`?

Reply:
For consistency, because `LOG_` is used everywhere else in this driver, I'd go with `LOG_ERR`.

This being said, I'm not quite sure about the correct choice for UART driver:
- It is true that LOG on uart driver is not often successful for an obvious reason
- On the other hand, as mentioned, __ASSERT is not always enabled (except on tests)
Anyway, f it has to be changed, it should be done globally to this driver as part of a new PR.